core/src/main/java/org/mule/runtime/core/internal/policy/CompositeOperationPolicy.java

```diff
@@ -97,8 +102,10 @@ public CoreEvent process(CoreEvent event) throws MuleException {
    */
   @Override
   protected Publisher<CoreEvent> processNextOperation(CoreEvent event) {
+    logOperationEvent(event.getContext(), event.getMessage());
```

Comment:
this "before" log should be done inside the publisher chain returned by this method